givenergy_modbus/client/client.py

```
@@ -213,12 +213,60 @@ async def _probe(self, request: TransparentRequest, timeout: float, retries: int
+    async def _detect_bcu_stacks(
+        self,
+        caps: PlantCapabilities,
+        prior: PlantCapabilities | None,
+        probe_timeout: float,
+        probe_retries: int,
+    ) -> None:
+        """Populate caps.bcu_stacks. Hinted mode trusts prior layout; cold mode reads BMS at 0xA0."""
+        if prior is not None:
+            # Hinted: probe each previously-seen BCU and record what the BCU actually
+            # reports for its module count (rather than trusting `prior`). The probe
+            # populates IR(60–64) into the register cache; IR(64) is the BCU's own
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)